src/test/java/org/assertj/core/api/junit/jupiter/SoftAssertionsExtension_InjectionSanityChecking_Test.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2020 the original author or authors.
+ */
+package org.assertj.core.api.junit.jupiter;
+
+import static org.assertj.core.api.junit.jupiter.TestKitUtils.assertThatTest;
+
+import java.util.Date;
+
+import org.assertj.core.api.AbstractSoftAssertions;
+import org.assertj.core.api.SoftAssertions;
+import org.assertj.core.api.SoftAssertionsProvider;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestInstance.Lifecycle;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.ExtensionConfigurationException;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+
+@DisplayName("SoftAssertionsExtension injection sanity checking test")
+class SoftAssertionsExtension_InjectionSanityChecking_Test {
+
+  @ExtendWith(SoftAssertionsExtension.class)
+  static private class TestBase {
+    @Test
+    void myTest() {}
+  }
+
+  class NoDefaultSoftAssertionsProvider extends AbstractSoftAssertions implements SoftAssertionsProvider {
+    public NoDefaultSoftAssertionsProvider(String param) {}
+  }
+
+  @Disabled("Run by the testkit")
+  static class NoDefaultConstructorTest extends TestBase {
+    @InjectSoftAssertions
+    NoDefaultSoftAssertionsProvider usp;
+  }
+
+  @Test
+  void field_with_no_default_constructor_throws_exception() {
+    assertThatTest(NoDefaultConstructorTest.class).isInstanceOf(ExtensionConfigurationException.class)
+                                                  .hasMessage("[usp] SoftAssertionsProvider [" + NoDefaultSoftAssertionsProvider.class.getName() + "] does not have a default constructor");
```

Comment:
minor: hasMessage supports string format syntax as in `hasMessage("%s is not a valid input", 12);`